File: benchmark/output/python-pygame/gemini/state_machine_game.py
import pygame
import sys
import random

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("3-State Target Clicker")
clock = pygame.time.Clock()
font = pygame.font.Font(None, 36)

state = 0
score = 0
timer = 10.0
target_x = random.uniform(0.0, 780.0)
target_y = random.uniform(0.0, 580.0)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN:
            if event.key == pygame.K_SPACE:
                if state == 0:
                    state = 1
                    score = 0
                    timer = 10.0
                    target_x = random.uniform(0.0, 780.0)
                    target_y = random.uniform(0.0, 580.0)
                elif state == 2:
                    state = 0
        elif event.type == pygame.MOUSEBUTTONDOWN:
            if state == 1 and event.button == 1:
                mx, my = event.pos
                if target_x <= mx <= target_x + 20.0 and target_y <= my <= target_y + 20.0:
                    score += 1
                    target_x = random.uniform(0.0, 780.0)
                    target_y = random.uniform(0.0, 580.0)

    if state == 1:
        timer -= dt
        if timer <= 0.0:
            timer = 0.0
            state = 2

    screen.fill((0, 0, 0))

    if state == 0:
        text = font.render("Press SPACE to Start", True, (255, 255, 255))
        screen.blit(text, (250.0, 280.0))
    elif state == 1:
        pygame.draw.rect(screen, (255, 255, 255), (int(target_x), int(target_y), 20, 20))
        
        score_text = font.render("Score: " + str(score), True, (255, 255, 255))
        timer_text = font.render(f"Timer: {max(0.0, timer):.1f}", True, (255, 255, 255))
        screen.blit(score_text, (10, 10))
        screen.blit(timer_text, (10, 40))
    elif state == 2:
        gameover_text = font.render("Game Over! Score: " + str(score), True, (255, 255, 255))
        restart_text = font.render("Press SPACE to Restart", True, (255, 255, 255))
        screen.blit(gameover_text, (250.0, 260.0))
        screen.blit(restart_text, (250.0, 300.0))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/gemini/state_machine_game.lua
local state = 0
local score = 0
local timer = 10.0
local target_x = 0
local target_y = 0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Target Clicker")
end

function love.update(dt)
    if state == 1 then
        timer = timer - dt
        if timer <= 0.0 then
            timer = 0.0
            state = 2
        end
    end
end

function love.draw()
    if state == 0 then
        love.graphics.print("Press SPACE to Start", 250, 280)
    elseif state == 1 then
        love.graphics.rectangle("fill", target_x, target_y, 20, 20)
        love.graphics.print("Score: " .. score, 10, 10)
        love.graphics.print(string.format("Timer: %.1f", timer), 10, 30)
    elseif state == 2 then
        love.graphics.print("Game Over! Score: " .. score, 250, 260)
        love.graphics.print("Press SPACE to Restart", 250, 300)
    end
end

function love.keypressed(key)
    if state == 0 then
        if key == "space" then
            state = 1
            score = 0
            timer = 10.0
            target_x = love.math.random(0, 780)
            target_y = love.math.random(0, 580)
        end
    elseif state == 2 then
        if key == "space" then
            state = 0
        end
    end
end

function love.mousepressed(x, y, button)
    if state == 1 then
        if button == 1 then
            if x >= target_x and x <= target_x + 20 and y >= target_y and y <= target_y + 20 then
                score = score + 1
                target_x = love.math.random(0, 780)
                target_y = love.math.random(0, 580)
            end
        end
    end
end